src/month.rs

```diff
@@ -195,8 +223,23 @@ pub struct Months(pub(crate) u32);
 
 impl Months {
     /// Construct a new `Months` from a number of months
+    pub fn new_opt(num: u32) -> Option<Self> {
+        if num <= core::i32::MAX as u32 {
```

Comment:
I'm not really a fan of this as the constraint ends up not being expressed in the type system, so we have to lean on convention/remembering/understanding. IMO it's better just to it as `u32` (which also makes the initialization API simple) and check the range in usage context.

Reply:
My thinking with this is that if the `Months` is invalid, its better for this to show up when creating the months rather than adding it, but admittedly most `Months` will be made and added as part of the same expression. Perhaps another option is taking `i32` here but returning `None` on negative values? Or else removing `new_opt` and having the check be done inside `overflowing_add/sub`. This also forces months to have one less possible value on the sub side due to `i32::MAX != abs(i32::MIN)`, but potentially this is acceptable as it is erring on the side of allowing a slightly smaller maximum subtraction?

Reply:
But that's just the point: a `Months::new(u32::MAX)` is not invalid by itself -- it's conceptually perfectly valid. It's only invalid in the context of what `checked_add_months()`/`checked_sub_months()` are trying to do, as such IMO the burden of validation should fall on the context, not the type itself.

Reply:
Makes sense, will move the checks there